streams/src/main/java/org/apache/kafka/streams/processor/DefaultPartitionGrouper.java

```diff
@@ -36,7 +36,11 @@
  * Join operations requires that topics of the joining entities are copartitoned, i.e., being partitioned by the same key and having the same
  * number of partitions. Copartitioning is ensured by having the same number of partitions on
  * joined topics, and by using the serialization and Producer's default partitioner.
+ *
+ * @deprecated since 2.4 release; will be removed in 3.0.0 via KAFKA-7785
  */
+// TODO: move this class into an internal package instead of deleting it (cf KAFKA-7785)
```

Comment:
Since we have a Jira ticket that is even referenced here, I would prefer to remove the ToDo from the code.

Reply:
What is bad about the `TODO` ?

Reply:
`TODO` pollute the code IMO. They tend to be forgotten or they describe something that is obsolete or even worse the description is not comprehensible. For all of these reasons, I try to avoid `TODO`s as much as I can. Using Jira tickets to document `TODO`s is much more effective and does not pollute code. Since we have already a Jira ticket that tells us what is needed to be done in this case, having a TODO is redundant. Up to you, whether you want to follow my reasoning or not. I already approved this PR.

Reply:
Ack.